### PR TITLE
fix(streams): integrate float32-safe scale bounds

### LINKED ISSUE #421: ScaleDriftStream accepts reversed/non-finite log-scale bounds (dead drift, plausible stream); DynamicScaleShiftStream accepts non-positive/non-finite scale bounds (NaN observations)
## Summary

Two synthetic streams in `alberta_framework/streams/synthetic.py` accept scale bounds that make the stream measure something other than what its docstring says, without any error:

1. **`ScaleDriftStream`** — "Non-stationary stream where feature scales drift via random walk … `min_log_scale`: Minimum log-scale (clips random walk)". `__init__` performs no validation. With reversed bounds `jnp.clip(x, lo, hi)` degenerates to `hi`, so every scale is pinned and the advertised drift never happens — but the emitted stream is entirely finite and looks normal. The class exists specifically to measure "whether an online normalizer (EMANormalizer) can track scale shifts faster than Autostep's `v_i`"; with reversed bounds the arm silently measures a stationary-scale problem and the A/B is meaningless. Non-finite log bounds produce NaN observations.
2. **`DynamicScaleShiftStream`** — `min_scale <= 0`, non-finite bounds, or `min_scale > max_scale` are accepted; `jnp.log(min_scale)` then seeds NaN/inf scales and NaN/inf observations from an accepted config. `make_scale_range`'s log branch was already hardened for exactly this (#256) with `_require_normal_float32_scale`; the stream constructor was not.

## Falsifier (deterministic, pristine `main` @ `4f75cc8`)

```
=== ScaleDriftStream accepts min_log_scale > max_log_scale ===
  step 0: log_scales=[-4. -4. -4. -4.]  obs=[ 0.0062  0.0189 -0.0062  0.0102]
  step 3: log_scales=[-4. -4. -4. -4.]  obs=[ 0.0300 -0.0149  0.0155 -0.0200]
  -> every log scale pinned to max_log_scale=-4.0; the configured drift is dead
=== ScaleDriftStream accepts non-finite log-scale bounds ===
  obs: [nan nan nan]  target: [-2.878634]
=== DynamicScaleShiftStream accepts min_scale<=0 / non-finite ===
  {'min_scale': 0.0}:   init scales=[nan nan nan] obs=[nan nan nan]
  {'min_scale': -1.0}:  init scales=[nan nan nan] obs=[nan nan nan]
  {'min_scale': nan}:   init scales=[nan nan nan] obs=[nan nan nan]
  {'max_scale': inf}:   init scales=[inf inf inf] obs=[ inf  inf -inf]
```

Expected: `ScaleDriftStream` requires finite log bounds with `min_log_scale <= max_log_scale`; `DynamicScaleShiftStream` requires normal-float32 positive scale bounds (the existing helper) with `min_scale <= max_scale`.

## Dedup

Adjacent, closed: #256 (`make_scale_range(log_spaced=True)` bounds), #143 (zero schedule moduli), #249 (`FrequencyMismatchStream` omega), #269 (`CompositionalStream` weight_scale), #173 (`XDistShiftStream`), #298 (`ScaledStreamWrapper` ndim). `gh issue/pr list --search "ScaleDriftStream" / "DynamicScaleShiftStream" / "min_log_scale"`: nothing; no open PR touches `streams/synthetic.py`.

Fix PR follows (failing-test-first).

### PR DESCRIPTION
**Pull request by lalalune**
Supersedes conflicting #422 while preserving its contributor commits and final repaired behavior on current `main`.

Both `ScaleDriftStream` and `DynamicScaleShiftStream` validate exact host values and their binary32 execution sinks, reject class-spoofed/non-finite/overflowing values, compare canonicalized bounds, and store the values JAX actually consumes. The integration retains newer synthetic-stream validation already on main.

Validation: `tests/test_streams.py` — 164 passed; Ruff clean. No outputs or evidence artifacts changed.

Closes #421.

AI assistance: OpenAI GPT-5.6 Codex desktop. Attribution: self-reported.

### CODE REVIEW

**Comment by lalalune** [comment]
Superseded by the deeper #620. #618 validated finite log bounds but still admitted finite float32 bounds whose exponential becomes zero or infinity; #620 closes the actual execution sink and preserves the original contributor commits.
